File: acoustic_abundance_estimates/Echoview_Processing.m
%% Utilize zoop/fish processing functions to generate excel files that can be imported/graphed in R
% This is a separate file so that the function can be put into a loop and iterated over months/days
% Author: Delphine Mossman
% Date Created: 12 Oct 2023
% Date Last Modified: 23 June 2025
% 1. Get the directory of exported Echoview data
% 2. For each file, run either the fish or zooplankton function


clear variables; close all; clc;

% Add paths to the Matlab processing scripts
addpath(genpath('C:\Users\Delphine\Box\messy_acoustics_processing\'))

%% Initialization

% Path to exported Echoview data
dep_name = uigetdir("C:\Users\Delphine\Box\Glider Data\", "Select the glider deployment folder.");
addpath(genpath(dep_name));
dep_name = char(extractBetween(dep_name,35,strlength(dep_name)));

% Some formatting things
dep_year = char(extractBetween(dep_name,6,9));

dep_start = input("Enter the numerical month the deployment started: ");
dep_end = input("Enter the numerical month the deployment ended: ");

%% Function loop

for i = dep_start:dep_end
    yr=string(dep_year);
    % Directory with Echoview csv files
    input_dir = strcat('C:\Users\Delphine\Box\Glider Data\',dep_name,'\Echoview CSV Export Files',...
        '\',char(yr),sprintf('%02d',i),'\');
    % Directory to output formatted files
    output_dir = strcat('C:\Users\Delphine\Box\Glider Data\',dep_name,'\Derived Biomass Data\');

    days = dir([input_dir '*_200kHz.csv']);
    
    for j = 1:length(days)
        mo = i;
        da = days(j).name(12:13);
        
        % Needs to be manually changed depending on whether the AZFP is
        % zoop or fish configured
        Echoview_to_zoop_biomass(yr,mo,da,input_dir,output_dir)
        % Echoview_to_fish_biomass(yr,mo,da,input_dir,output_dir)

    end
end

File: acoustic_abundance_estimates/Echoview_Processing.R
# Setup

rm(list = ls())

library(tidyverse)
library(tcltk)

source("./AZFP_acoustics_processing/misc/choose_directory.R")
source("./AZFP_acoustics_processing/misc/create_dir.R")
source("./AZFP_acoustics_processing/acoustic_abundance_estimates/echoview_to_abundance_functions.R")

# Getting file paths set up

glider_dep = choose_directory() %>% substring(., regexpr("ru[0-9]{2}-*", .))
year = substr(glider_dep,6,9)

output_dir = paste0("C:/Users/dm1679/Box/Glider Data/",
                    glider_dep,
                    "/Derived Biomass Data/"
                    ,"School Detection Tests/"
)
create_dir(output_dir)

dep_start = as.numeric(readline("Enter the numerical month the deployment started: "))
dep_end = as.numeric(readline("Enter the numerical month the deployment ended: "))

# For each month
for(i in dep_start:dep_end) {
  
  # Get input folder and list of files
  input_dir = paste0("C:/Users/dm1679/Box/Glider Data/",
                     glider_dep,
                     "/Echoview CSV Export Files/",
                     "School Detection Tests/",
                     year,sprintf("%02d",i),"/")
  
  days = list.files(input_dir, pattern = "*_abundance_median.csv")
  # days = list.files(input_dir, pattern = "*200kHz.csv")
  
  if(length(days) == 0){
    next
  }
  
  # For each day
  for(j in 1:length(days)) {
    mo = i
    da = as.numeric(substr(days[j], 12, 13))
    
    # Calculate the abundance metrics
    echoview_to_zoop_abundance_aggregations(year, mo, da, input_dir, output_dir)
    # echoview_to_zoop_abundance(year, mo, da, input_dir, output_dir)
    # echoview_to_fish_class(year, mo, da, input_dir, output_dir)
  }
}
